### PR TITLE
Project/27 detekt

### PR DESCRIPTION
**Pull request by kidinov**
Epic branch of detekt integration. All code was reviewed already except this commit https://github.com/woocommerce/woocommerce-android/commit/1ca2c4176916c0b9e261b6bdb58ba99e8d9f8f08

### CODE REVIEW

File: config/checkstyle.xml [hunk 1]
```diff
@@ -6,34 +6,7 @@
...
-    <!-- Checks for Size Violations.                    -->
-    <!-- See http://checkstyle.sf.net/config_sizes.html -->
-    <!--
-    <module name="FileLength"/>
-    -->
-
-    <!-- Checks for whitespace                               -->
-    <!-- See http://checkstyle.sf.net/config_whitespace.html -->
-    <module name="FileTabCharacter"/>
+    <property name="fileExtensions" value="kt"/>
```
Comment (wzieba) [inline config/checkstyle.xml:9]: If we disable `checkstyle` for java, what are the advantages of leaving `checkstyle` at all? Are those additional checks needed if we want to follow official guidelines?

**Review by wzieba — APPROVED** [review]
All looks good to me, just one question.

Thanks for working on this!

File: config/checkstyle.xml (hunk 1 above)
Comment (kidinov) [reply config/checkstyle.xml:9]: Totally agree. It was a request to keep it for now in the P2 post where we started to discuss all of this. It doesn't hurt us. Eventually, we could replace this check (if needed) with custom detekt rules

Comment (wzieba) [reply config/checkstyle.xml:9]: Gotcha, thanks.

> Eventually, we could replace this check (if needed) with custom detekt rules

Yeah, that's something I'd prefer too - less tools, the better :)

**Pull request merged**